source/world_builder/utilities.cc

```diff
@@ -427,9 +434,166 @@ namespace WorldBuilder
+          // intersect the trench given the obliquity_vector.
+          obliquity_vector = obliquity_vector / obliquity_vector.norm();
+          if (!std::isnan(closest_point_on_line_2d[0]))
+            {
+              Point<2> check_point_surface_2d_temp_ob = check_point_surface_2d;
+
+              if (!bool_cartesian)
+                {
+                  const double normal = std::fabs(point_list[i_section_min_distance+static_cast<size_t>(std::round(fraction_CPL_P1P2))][0]-check_point_surface_2d[0]);
+                  const double plus   = std::fabs(point_list[i_section_min_distance+static_cast<size_t>(std::round(fraction_CPL_P1P2))][0]-(check_point_surface_2d[0]+2*Consts::PI));
+                  const double min    = std::fabs(point_list[i_section_min_distance+static_cast<size_t>(std::round(fraction_CPL_P1P2))][0]-(check_point_surface_2d[0]-2*Consts::PI));
+
+                  // find out whether the check point, checkpoint + 2pi or check point -2 pi is closest to the point list.
+                  if (plus < normal)
+                    {
+                      check_point_surface_2d_temp_ob[0]+= 2*Consts::PI;
+                    }
+                  else if (min < normal)
+                    {
+                      check_point_surface_2d_temp_ob[0]-= 2*Consts::PI;
+                    }
+                }
+
+              const Point<2> AB_normal_ob = closest_point_on_curve.normal*closest_point_on_line_2d.distance(reference_point);//*AB.norm();
+              const Point<2> local_reference_point_ob = AB_normal_ob*1.+closest_point_on_line_2d;
+              const bool reference_normal_on_side_of_line_ob =  (closest_point_on_line_2d-local_reference_point_ob).norm_square() < (check_point_surface_2d_temp_ob-local_reference_point_ob).norm_square();
+              const bool reference_point_on_side_of_line_ob =  (point_list[point_list.size()-1][0] - point_list[0][0])*(reference_point[1] - point_list[0][1]) - (reference_point[0] - point_list[0][0])*(point_list[point_list.size()-1][1] - point_list[0][1]) < 0.;
+
+              double line_factor;
+              double old_dist;
+              double new_dist;
+              Point<2> iterable_check_point_surface_2d = check_point_surface_2d;
+              Objects::ClosestPointOnCurve iterable_closest_point_on_curve = bezier_curve.closest_point_on_curve_segment(iterable_check_point_surface_2d);
+              bool converged = false;
+
+              if (reference_normal_on_side_of_line_ob != reference_point_on_side_of_line_ob)
+                {
+                  // We want to take the current checkpoint and move it along the obliquity vector using a parameterized line
+                  // towards the bezier curve until we find a point on the bezier curve that intersects this parameterized line.
+                  for (unsigned int i = 0; i < 100; ++i)
```

Comment:
why 100? What if we don't converge before?

Reply:
Yea I don't really have a good reason for picking 100, I was torn between making the number of iterations an input parameter or not. I just chose 100 as a starting number, but if you think it should be higher/a user input parameter I would be ok with changing this.